modules/EvseManager/EvseManager.cpp

```diff
@@ -1395,6 +1405,38 @@ static double get_cable_check_voltage(double ev_max_cpd, double evse_max_cpd) {
     return cable_check_voltage;
 }
 
+double EvseManager::get_over_voltage_threshold() {
+
+    float ev_max_voltage = 500.;
+
+    if (ev_info.maximum_voltage_limit.has_value()) {
+        ev_max_voltage = ev_info.maximum_voltage_limit.value();
+    } else {
+        EVLOG_error << "OverVoltageThreshold: Did not receive EV maximum voltage, falling back to 500V";
+    }
+
+    auto evse_caps = get_powersupply_capabilities();
+
+    double negotiated_max_voltage = std::min(ev_max_voltage, evse_caps.max_export_voltage_V);
+
+    double ovp = 550.;
+
+    // IEC 61851-23 (2023) 6.3.1.106.2 Table 103
+    if (negotiated_max_voltage > 500) {
+        ovp = 825.;
+    }
+
+    if (negotiated_max_voltage > 750) {
+        ovp = 935.;
+    }
+
+    if (negotiated_max_voltage > 850) {
+        ovp = 1100.;
+    }
```

Comment:
This could be an `if`/`else if` chain, beginning with the highest value.

Reply:
changed